sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADConditions.java

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.aad;
+
+import org.springframework.boot.autoconfigure.condition.AnyNestedCondition;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingClass;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.server.resource.BearerTokenAuthenticationToken;
+
+/**
+ * Conditions for activating AAD beans.
+ */
+public final class AADConditions {
+
+    public static final String BEARER_TOKEN_AUTHENTICATION_TOKEN_CLASS_NAME =
+        "org.springframework.security.oauth2.server.resource.BearerTokenAuthenticationToken";
+
+
+    /**
+     * Web application, web resource server or all in scenario condition.
+     */
+    static final class ClientRegistrationCondition extends AnyNestedCondition {
```

Comment:
How about do like this:
```
public final class AADConditions {
    private static boolean clientIdConfigured() {

    }

    private static boolean isWebApplication() {

    }

    private static boolean isResourceServer() {

    }

    private static boolean isResourceServerWithOBOFunction() {

    }

    private static boolean isWebApplicationAndResourceServer() {

    }

    static final class ClientRegistrationCondition implements Condition {

        @Override
        public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
            return clientIdConfigured() &&
                (isWebApplication() || isResourceServerWithOBOFunction() || isWebApplicationAndResourceServer());
        }
    }
|
```

Reply:
It seems simpler, but this way will not apply the condition built-in function, we need to use Classs.forName to check, lots of `ConditionOutcome` results will be set,  and AAD B2C side also uses the same pattern to extend the class AnyNestedCondition.

Reply:
> his way will not apply the condition built-in function

Maybe it's OK not to apply the built-in function.

> we need to use Classs.forName to check

It's not a problem.

> lots of ConditionOutcome results will be set

What does this mean?

> and AAD B2C side also uses the same pattern to extend the class AnyNestedCondition.

Then change AAD B2C together.

Reply:
I will use `SpringBootCondition` to extend the condition, then lots of methods can be reused.